src/main/kotlin/uk/gov/gdx/datashare/controller/BaseDataController.kt

```diff
@@ -0,0 +1,303 @@
+  suspend fun deleteConsumer(
+    @Schema(description = "Consumer ID", required = true)
+    @PathVariable id: UUID,
+  ) = consumerRepository.deleteById(id)
+  
+  @GetMapping("/consumerSubscriptions")
+  @Operation(
+    summary = "Get Consumer Subscriptions",
+    description = "Need scope of pubsub/maintain",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Consumer Subscriptions"
+      )
+    ]
+  )
+  suspend fun getConsumerSubscriptions() = consumerSubscriptionRepository.findAll().toList()
+
+  @DeleteMapping("/consumerSubscriptions/{id}")
+  @Operation(
+    summary = "Delete Consumer Subscription",
+    description = "Need scope of pubsub/maintain",
+    responses = [
+      ApiResponse(
+        responseCode = "204",
+        description = "Consumer Subscription deleted"
+      )
+    ]
+  )
+  suspend fun deleteConsumerSubscription(
+    @Schema(description = "Consumer Subscription ID", required = true)
+    @PathVariable id: UUID,
+  ) = consumerSubscriptionRepository.deleteById(id)
+  
+  @GetMapping("/egressEvents")
+  @Operation(
+    summary = "Get Egress Events",
+    description = "Need scope of pubsub/maintain",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Egress Event"
+      )
+    ]
+  )
+  suspend fun getEgressEvents() = egressEventDataRepository.findAll().toList()
+  
+  @DeleteMapping("/egressEvents/{id}")
+  @Operation(
+    summary = "Delete Egress Event",
+    description = "Need scope of pubsub/maintain",
+    responses = [
+      ApiResponse(
+        responseCode = "204",
+        description = "Event deleted"
+      )
+    ]
+  )
+  suspend fun deleteEgressEvent(
+    @Schema(description = "Event ID", required = true)
+  @PathVariable id: UUID,
+  ) = egressEventDataRepository.deleteById(id)
+
+  @GetMapping("/eventDataset")
+  @Operation(
+    summary = "Get Egress Datasets",
+    description = "Need scope of pubsub/maintain",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Egress Dataset"
+      )
+    ]
+  )
+  suspend fun getEgressDatasets() = eventDatasetRepository.findAll().toList()
+
+  @DeleteMapping("/eventDataset/{id}")
+  @Operation(
+    summary = "Delete Egress Event",
+    description = "Need scope of pubsub/maintain",
+    responses = [
+      ApiResponse(
+        responseCode = "204",
+        description = "Event deleted"
+      )
+    ]
+  )
+  suspend fun deleteEventDataset(
+    @Schema(description = "Event Dataset ID", required = true)
+    @PathVariable id: String,
+  ) = eventDatasetRepository.deleteById(id)
+
+  @GetMapping("/ingressEvent")
+  @Operation(
+    summary = "Get Ingress Events",
+    description = "Need scope of pubsub/maintain",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Ingress Event"
+      )
+    ]
+  )
+  suspend fun getIngressEvents() = ingressEventDataRepository.findAll().toList()
```

Comment:
Do you need the toList?  Return a flow instead?

Reply:
removed the to list :)